### PR TITLE
[FIX] Custom status fixes

### PR DESCRIPTION
**Pull request by wreiske**
Here are some fixes for the status message presence issues and other cleanup related to #13933

### CODE REVIEW

**Comment by wreiske** [comment]
Things that need to be addressed:

image

+ Setting status with /status slash command is restricted to the `edit-other-user-info` permission.

**Comment by wreiske** [comment]
There's still some serious weirdness here... if someone is offline and you look at their page, the top nav bar will show "offline" but the user info will actually show the correct status. 

I'm 💤 and this code makes me 🤢 so hopefully someone else can take a peek and figure out what's going wrong.

File: app/ui/client/components/header/headerRoom.js [hunk 1]
```diff
@@ -28,6 +28,11 @@ const getUserStatus = (id) => {
 	return roomTypes.getUserStatus(roomData.t, id) || 'offline';
 };
 
+const getUserStatusText = (id) => {
+	const roomData = Session.get(`roomData${ id }`);
+	return roomTypes.getUserStatusText(roomData.t, id) || 'offline';
```
Comment (Hudell) [inline app/ui/client/components/header/headerRoom.js]: This will cause the status to be shown as offline if there's no custom message.

Comment (wreiske) [reply app/ui/client/components/header/headerRoom.js]: Weird... My code doesn't have this...
image

Shows commmited, too.

Comment (wreiske) [reply app/ui/client/components/header/headerRoom.js]: https://github.com/RocketChat/Rocket.Chat/pull/14853/commits/31137ac1907a2d9cb59bf66019abe8cb73f456ef#diff-a89db19a330b54652f42b8fa6cc29bcdR32

Comment (Hudell) [reply app/ui/client/components/header/headerRoom.js]: Sorry, my mistake. I think I was looking at a specific commit when I commented.

File: app/ui-sidenav/client/sidebarHeader.js [hunk 2]
```diff
@@ -316,18 +316,24 @@ Template.sidebarHeader.events({
...
+				'away',
+				'busy',
+			];
 			const userStatusList = Object.keys(userStatus.list).map((key) => {
 				const status = userStatus.list[key];
-				const customName = status.localizeName ? null : status.name;
 				const name = status.localizeName ? t(status.name) : status.name;
 				const modifier = status.statusType || user.status;
+				const defaultStatus = STATUS_MAP.includes(status.id);
+				const statusText = defaultStatus ? '' : user.statusText || name || t(user.status);
```
Comment (Hudell) [inline app/ui-sidenav/client/sidebarHeader.js]: `name` should come before `user.statusText` here, or it breaks the admin-defined extra status.

Comment (Hudell) [reply app/ui-sidenav/client/sidebarHeader.js]: Oh, but then it causes regular statuses to overwrite the custom ones. We'll need to change the logic here a bit.

Comment (Hudell) [reply app/ui-sidenav/client/sidebarHeader.js]: ```const statusText = defaultStatus ? null : status.name;```

**Comment by wreiske** [comment]
@Hudell the top header still shows "offline" even though the status is set to "on vacation, I'll be back 7/1"
image

**Comment by Hudell** [comment]
Ah, that's right. When you refresh, the client's User collection is empty so it doesn't have information about offline users. The user panel knows the right status because it loads the user data.

I'll see what I can do to fix this.

**Comment by wreiske** [comment]
> Ah, that's right. When you refresh, the client's User collection is empty so it doesn't have information about offline users. The user panel knows the right status because it loads the user data.
> 
> I'll see what I can do to fix this.

I was looking at how the side panel does it,
```
this.subscribe('fullUserData', username, 1, () => this.loadingUserInfo.set(false));
```

but then thought maybe it would be better to use the subscription API to pull it? I'll let you fix it! I've got other things to work on. Thanks!

**Comment by wreiske** [comment]
WIth the latest fixes, there seems to be an issue when changing the status to one of the pre-defined custom statuses.

image


If I click on "Out To Lunch" which should make me away, and then select "In a meeting" that should make me busy, it only changes me to busy and doesn't change the status to "in a meeting."

We should be able to change to one of the default statuses (offline, away, busy, online...) no matter what the current status text is.

If my status is "Working on some cool stuff" and I initially set it to "busy", I should be able to change my "busy" to "online' without my status changing. 

If my status is "PTO (out of office, PTO)" I should be able to then change it to a pre-defined status of "In A Meeting" and it should change to the pre-defined status, HOWEVER, if I just click on "online" instead of "away", my status should stay "PTO" until I manually change it to something else.

**Comment by Hudell** [comment]
I was unable to replicate the problem here. All status changes worked fine for both text and presence.

**Comment by wreiske** [comment]
> I was unable to replicate the problem here. All status changes worked fine for both text and presence.

Yup! I can't reproduce it either. Weird. maybe I was in a weird state?... I say this commit is good to go!!!! @Hudell @sampaiodiego

**Pull request merged**